src/main/java/com/epam/esm/service/impl/TagServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package com.epam.esm.service.impl;
+
+import com.epam.esm.dao.DaoException;
+import com.epam.esm.dao.TagDao;
+import com.epam.esm.model.Tag;
+import com.epam.esm.service.CertificateServiceException;
+import com.epam.esm.service.TagService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.stereotype.Service;
+import org.springframework.util.MultiValueMap;
+
+
+import java.util.List;
+@Service
+public class TagServiceImpl implements TagService {
+
+    private TagDao tagDao;
+    @Autowired
+    public TagServiceImpl(TagDao tagDao) {
+        this.tagDao = tagDao;
+    }
+
+    @Override
+    public List<Tag> getByParameters(MultiValueMap<String, String> params) throws DaoException {
```

Comment:
I think it's better to add separate method for findAll.

Reply:
Fixed method, now it works like in Certificate with an ability to get a tag by a part of name.